cranelift/codegen/src/isa/riscv64/lower.isle

```diff
@@ -1214,19 +1214,126 @@
 
 
 ;;;; Rules for `rotl` ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
-(rule (lower (has_type (fits_in_64 ty) (rotl x y)))
-  (lower_rotl ty (zext x) (value_regs_get y 0)))
+
+(rule 0 (lower (has_type (fits_in_64 ty) (rotl rs amount)))
+  (let
+    ((rs XReg (zext rs))
+      (amount XReg (value_regs_get amount 0))
+      (x ValueRegs (gen_shamt ty amount))
+      (shamt XReg (value_regs_get x 0))
+      (len_sub_shamt Reg (value_regs_get x 1))
+      ;;
+      (part1 Reg (rv_sll rs shamt))
+      ;;
```

Comment:
Empty comments?

Reply:
[Don't shoot the messenger](https://github.com/bytecodealliance/wasmtime/blob/ad4f923bb7f6b6fe09f9282dbf80b711542c372a/cranelift/codegen/src/isa/riscv64/inst.isle#L2232-L2234) -- but yeah I'll go through and clean this up